samples/CMakeLists.txt

```diff
@@ -71,11 +71,11 @@ if (CONFIG_SPM)
   endif()
 endif()
 
-if (CONFIG_BUILD_WITH_TFM AND CONFIG_UART_1_NRF_UARTE)
+if (CONFIG_TFM_SECURE_UART1 AND CONFIG_UART_1_NRF_UARTE)
   message(FATAL_ERROR "
-    Non-secure applications built with TF-M cannot use UART1, because this
-    instance is used by the TF-M secure application. Disable the uart1 node
-    in devicetree. See the following file:
+    Non-secure applications cannot use UART1 when TFM_SECURE_UART1 is enabled,
+    because this instance is used by the TF-M secure application.
+    Disable the uart1 node in devicetree. See the following file:
     nrf/samples/tfm/tfm_hello_world/boards/nrf9160dk_nrf9160_ns.overlay
     for an example how to do it.")
 endif()
```

Comment:
Is this really needed ?

Why not do:
```
config TFM_SECURE_UART1
	bool "Use UART1 for TF-M logging"
	depends on !UART_1_NRF_UARTE
```
in Kconfig, and then this whole warning can go away.

```suggestion
```

Reply:
I think that this is helps with user experience to explain a bit more thoroughly what they should do. The depends can be added though.

Reply:
with the depends they can never get into this situation in the first place.

Secondly, a CMake error prevents the user from entering menuconfig to solve the problem.
See also: https://github.com/zephyrproject-rtos/zephyr/issues/9573

Reply:
That's is a very good point actually, I will change accordingly!